Source/Core/VideoCommon/TextureDecoder.h

```diff
@@ -14,94 +14,101 @@ enum
-
-  _GX_TF_CTF = 0x20,  // flag for copy-texture-format only (used internally by dolphin)
-
-  // These are extra formats that can be used when copying from efb,
-  // they use one of texel formats from above, but pack diffrent data into them.
-  GX_CTF_R4 = 0x0 | _GX_TF_CTF,
-  GX_CTF_RA4 = 0x2 | _GX_TF_CTF,
-  GX_CTF_RA8 = 0x3 | _GX_TF_CTF,
-  GX_CTF_YUVA8 = 0x6 | _GX_TF_CTF,  // YUV 4:4:4 - Dolphin doesn't implement this format as no
-                                    // commercial games use it
-  GX_CTF_A8 = 0x7 | _GX_TF_CTF,
-  GX_CTF_R8 = 0x8 | _GX_TF_CTF,
-  GX_CTF_G8 = 0x9 | _GX_TF_CTF,
-  GX_CTF_B8 = 0xA | _GX_TF_CTF,
-  GX_CTF_RG8 = 0xB | _GX_TF_CTF,
-  GX_CTF_GB8 = 0xC | _GX_TF_CTF,
-
-  // extra depth texture formats that can be used for efb copies.
-  GX_CTF_Z4 = 0x0 | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z8H = 0x8 | _GX_TF_ZTF | _GX_TF_CTF,  // This produces an identical result to to GX_TF_Z8
-  GX_CTF_Z8M = 0x9 | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z8L = 0xA | _GX_TF_ZTF | _GX_TF_CTF,
-  GX_CTF_Z16R = 0xB | _GX_TF_ZTF | _GX_TF_CTF,  // Reversed version of GX_TF_Z16
-  GX_CTF_Z16L = 0xC | _GX_TF_ZTF | _GX_TF_CTF,
+  // These values represent texture format in GX registers.
+  I4 = 0x0,
+  I8 = 0x1,
+  IA4 = 0x2,
+  IA8 = 0x3,
+  RGB565 = 0x4,
+  RGB5A3 = 0x5,
+  RGBA8 = 0x6,
+  C4 = 0x8,
+  C8 = 0x9,
+  C14X2 = 0xA,
+  CMPR = 0xE,
 };
 
-enum TlutFormat
+static inline bool IsColorIndexed(TextureFormat format)
 {
-  GX_TL_IA8 = 0x0,
-  GX_TL_RGB565 = 0x1,
-  GX_TL_RGB5A3 = 0x2,
+  return format == TextureFormat::C4 || format == TextureFormat::C8 ||
+         format == TextureFormat::C14X2;
+}
+
+// The EFB Copy pipeline looks like:
+//
+//   1. Read EFB -> 2. Select color/depth -> 3. Downscale (optional)
+//   -> 4. YUV conversion (optional) -> 5. Encode Tiles -> 6. Write RAM
+//
+// The "Encode Tiles" stage receives RGBA8 texels from previous stages and encodes them to various
+// formats. EFBCopyFormat is the tile encoder mode. Note that the tile encoder does not care about
+// color vs. depth or intensity formats - it only sees RGBA8 texels.
+enum class EFBCopyFormat
+{
+  // These values represent EFB copy format in GX registers.
+  // Most (but not all) of these values correspond to values of TextureFormat.
+  R4 = 0x0,  // R4, I4, Z4
+
+  // FIXME: Does 0x1 (Z8) have identical results to 0x8 (Z8H)?
+  //        Is either or both of 0x1 and 0x8 used in games?
+  R8_0x1 = 0x1,  // R8, I8, Z8H (?)
+
+  RA4 = 0x2,  // RA4, IA4
+
+  // FIXME: Earlier versions of this file named the value 0x3 "GX_TF_Z16", which does not reflect
+  //        the results one would expect when copying from the depth buffer with this format.
+  //        For reference: When copying from the depth buffer, R should receive the top 8 bits of
+  //                       the Z value, and A should be either 0xFF or 0 (please investigate).
+  //        Please test original hardware and make sure dolphin-emu implements this format
+  //        correctly.
```

Comment:
Actually, the current behavior makes perfect sense now.

The Red channel contains bits Z[23:16] while the Alpha channel contains 0xff.

So formats like RA8 or RA4 will have the upper bits of Z in the red channel and all ones in the alpha channel.